File: AnyCode/MaxSubarray.c
#include <stdio.h>

int* find_max_crossing_subarray(int* A, int low, int mid, int high)
{
    int max_left, max_right;
    
    // find max subarray in left part;
    int left_sum = -65435, sumL = 0;
    for(int i = mid; i >= low; i--){
        sumL = sumL+A[i];
        if(sumL>left_sum){
            left_sum = sumL;
            max_left = i;
        }
    }
    
    // find max subarray in right part;
    int right_sum = -65435, sumR = 0;
    for(int j = mid+1; j <= high; j++){
        sumR = sumR+A[j];
        if(sumR>right_sum){
            right_sum = sumR;
            max_right = j;
        }
    }
    
    static int B[] = {max_left, max_right, left_sum+right_sum};

    for(int i = 0; i <= 2; i++){
        printf("%d ", B[i]);
    }
    printf("\n");
    
    return B;
}

int main(int argc, char** argv)
{
    int array[] = {};
    find_max_crossing_subarray(array, 0, 4, 8);
    

}

File: AnyCode/MaxSubarray.py
def find_max_crossing_subarray(A):

    low = 0; 
    mid = (len(A)//2)-1;
    high = len(A);

    left_sum = -6543543486;
    sum = 0;
    for i in list(range(low, mid+1))[::-1]:
        sum = sum+A[i];
        if(sum>left_sum):
            left_sum = sum;
            max_left = i;       ## 1-7: Find max subarray in left part;
            
    right_sum = -6543543486;
    sum = 0;
    for j in range(mid+1, high):
        sum = sum+A[j];
        if(sum>right_sum):
            right_sum = sum;
            max_right = j;      ## 11-17: Find max subarray in right part;

    return max_left, max_right, left_sum+right_sum;



def find_max_subarray(A, low, high):
    if(high == low):
        return low, high, A[low];

    mid = (low+high)//2;

    else:
        
        ll, lh, ls = find_max_subarray(A, low, mid);
        rl, rh, rs = find_max_subarray(A, mid+1, high);
        cl, ch, cs = find_max_crossing_subarray(A);
        
        if(ls >= rs and ls > cs):
            return ll, lh, ls;
            
        elif(rs >= ls and ls >= cs):
            return rl, rh, rs;
            
        else:
            return cl, ch, cs;
            

w = find_max_crossing_subarray([-541,-1458,-15,-15,-51, -8465]);
print(w);